Automattic/wp-calypso
# Global Sidebar - update color scheme and show icons

Related to # https://github.com/Automattic/dotcom-forge/issues/6143, https://github.com/Automattic/dotcom-forge/issues/6144, https://github.com/Automattic/dotcom-forge/issues/6145, https://github.com/Automattic/dotcom-forge/issues/6146

## Proposed Changes

* update the color scheme per designs linked in issue
* add previously used icons back in place.
* migrates sidebar styles to a consistent use of `--color-sidebar-*` variables, so switching color schemes in the future should be a bit easier. Going back to our previous dark mode now should only requires changing a few variable definitions. (do I hear a dark mode toggle coming the future? 👂 😆  )

<img width="481" alt="Screenshot 2024-03-22 at 1 44 08 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/e0dbefba-a57b-4f23-b98d-02534cbdcdd1">


## Testing Instructions



* Load the calypso live link and test the global nav around reader, /me, and /sites.  Global site view is also a concern functionality wise, but this is only seen by early access users and is due to disappear with the new tab approach.
* Sidebar should continue to work as expected.
* Verify icons are in place.
* Test hover, focus, selected styles, as well as for submenu groups like in the reader.

BONUS
- this is not a requirement/goal of this PR, but we are in a much better state to change colors in the future (or adapt toggles / color schemes).
- Test changing the color variables under the global sidebar selectors at the top of client/mysites/sidebar/style.scss .. to change the background, text colors, hover colors, icon colors etc.
- You should be able to make a custom color scheme, or swap back to dark mode scheme, simply by only changing these variables.
- known limitations at this point - we don't have a specific color for hover state of icons - when selected they take on the selected text color, when hovered they use the unselected text color. Also, the WordPress.com link in the header is currently set as a background image - if we instead used the SVG we could have that take different colors from css as well. Those seem like the last things that we would need to fix if we wanted to supply a more broad selection of color schemes to users.

## Pre-merge Checklist
